cpg_pipes/types.py

```diff
@@ -0,0 +1,224 @@
+"""
+Wrappers for bioinformatics file types (CRAM, GVCF, FASTQ, etc).
+"""
+
+import logging
+from dataclasses import dataclass
+from enum import Enum
+from typing import List, Union
+
+from . import Path, to_path
+from . import utils
+from hailtop.batch import ResourceGroup, ResourceFile, Batch
+
+logger = logging.getLogger(__file__)
+
+
+class CramPath:
+    """
+    Represents alignment data on a bucket within the pipeline.
```

Comment:
Do all BAM / CRAMs have indexes? Is this a procedural guarantee? Are there ever cases where this might not be the case?

Reply:
Responded in the other PR, but in the current pipelines, that's guaranteed. Maybe in the future I'll support other cases